sql: logic test failure under high verbosity

A logic test is brittle and gives a different result under high verbosity.

Running
`make testlogic TESTFLAGS='-verbosity=10 -show-sql -test.v' TESTTIMEOUT=1h FILES=show_trace`

Produces:
```
           --- FAIL: TestLogic/local/show_trace/autocommit (0.01s)
            	logic.go:2228: 
            		 
            		testdata/logic_test/show_trace:462: SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE e'%1 CPut, 1 BeginTxn, 1 EndTxn%'
            		expected:
            		    r1: sending batch 1 CPut, 1 BeginTxn, 1 EndTxn to (n1,s1):1
            		    1 CPut, 1 BeginTxn, 1 EndTxn
            		    
            		but found (query options: "") :
            		    r1: sending batch 1 CPut, 1 BeginTxn, 1 EndTxn to (n1,s1):1
            		    1 CPut, 1 BeginTxn, 1 EndTxn
            		    proposing command 3b7f33574bc22715: 1 CPut, 1 BeginTxn, 1 EndTxn
            		    
```

--- a/pkg/sql/logictest/testdata/logic_test/show_trace
+++ b/pkg/sql/logictest/testdata/logic_test/show_trace
@@ -459,7 +459,7 @@ SET tracing = on; INSERT INTO t.kv3 (k, v) VALUES (1,1); SET tracing = off
 # We look for rows containing a BeginTxn and an EndTxn, as proof that the
 # insertNode is committing the txn.
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE e'%1 CPut, 1 BeginTxn, 1 EndTxn%'
+SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE e'%1 CPut, 1 BeginTxn, 1 EndTxn%' AND message NOT LIKE e'%proposing command%'
 ----
 r1: sending batch 1 CPut, 1 BeginTxn, 1 EndTxn to (n1,s1):1
 1 CPut, 1 BeginTxn, 1 EndTxn

